### PR TITLE
Support encrypted env file in secrets

### PR DESCRIPTION
**Pull request by lox**
Currently the README mentions that you can have an encrypted `env` file in your `SecretsBucket` which will be sourced and executed as part of the Buildkite environment hook.

### CODE REVIEW

**Comment by toolmantim** [comment]
What's this env thing?

**Comment by lox** [comment]
Updated the description @toolmantim. Because you might have different projects running on the one stack, you need a way to deliver ENV's.

File: packer/conf/hooks/environment [hunk 1]
```diff
@@ -19,11 +14,21 @@ if [[ -n "${BUILDKITE_SECRETS_BUCKET:-}" ]] && [[ -z "${BUILDKITE_SECRETS_KEY:-}
...
-# startup an ssh-agent with the ssh-key
 eval "$(ssh-agent -s)"
-ssh-add <(download_github_key "$SSH_KEY_NAME")
+
+if [[ -n "${BUILDKITE_SECRETS_BUCKET:-}" ]] ; then
+  [[ -d ~/secrets ]] || mkdir -p ~/secrets
+
+  echo "Downloading ssh key from $BUILDKITE_SECRETS_BUCKET"
+  if s3_download "${SSH_KEY_NAME:-id_rsa_github}" ~/secrets/id_rsa_github ; then
+    ssh-add ~/secrets/id_rsa_github
```
Comment (Vektah) [inline packer/conf/hooks/environment]: What was the reason to move away from <()?

It seems a little nicer then having to manually clean up these files.

**Comment by toolmantim** [comment]
Oh, the `env` technique is a good one

File: packer/conf/hooks/environment (hunk 1 above)
Comment (lox) [inline packer/conf/hooks/environment]: Yeah, agree @Vektah

**Pull request merged**